docs/docs/typechecking-with-proptypes.md

```diff
@@ -0,0 +1,157 @@
+---
+id: typechecking-with-proptypes
+title: Typechecking With PropTypes
+permalink: docs/typechecking-with-proptypes.html
+---
+
+As your app grows, you can catch a lot of bugs with typechecking. For some applications, you can use JavaScript extensions like [Flow](https://flowtype.org/) or [TypeScript](https://www.typescriptlang.org/) to typecheck your whole application. But even if you don't use those, React has some built-in typechecking abilities. To run typechecking on the props for a component, you can assign the special `propTypes` property:
+
+```javascript
+class Greeting extends React.Component {
+  render() {
+    return (
+      <h1>Hello, {this.props.name}</h1>
+    );
+  }
+}
+
+Greeting.propTypes = {
+  name: React.PropTypes.string
+};
+```
+
+`React.PropTypes` exports a range of validators that can be used to make sure the data you receive is valid. In this example, we're using `React.PropTypes.string`. When an invalid value is provided for a prop, a warning will be shown in the JavaScript console. For performance reasons, `PropTypes` is only checked in development mode.
```

Comment:
Nit:

>`PropTypes` is only checked in development mode.

Maybe say `propTypes` here? I'd like us to consistently use either `React.PropTypes` when talking about validators, or `propTypes` when talking about the field name, so that people don't attempt to define `PropTypes` pascal case property.